cmd/libsnap-confine-private/infofile.c

```diff
@@ -76,6 +82,31 @@ int sc_infofile_get_key(FILE *stream, const char *key, char **value, sc_error **
         }
         /* Replace the trailing newline character with the NUL byte. */
         line_buf[nread - 1] = '\0';
+
+        /* Handle ini sections (if requested via non-null section name) */
+        if (section != NULL) {
+            if (line_buf[0] == '[') {
+                section_matched = false;
+                char *sec_endptr = memchr(line_buf, ']', nread);
```

Comment:
Technically you want to look for the first `]` after the initial `[`. I would re-arrange the helper variables to that you have a pointer to `start_section_name` and `end_section_name`. Then you can `*end_section_name = '\0';` and `sc_streq(start_section_name, section)` below. This would be nicer in case you want to allow leading spaces or anything like that. It would also avoid the need to calculate `line_buf + 1` in unison with `line_buf[0]` above.

Reply:
Done (not sure if I followed your idea to the letter, but hopefully it's makes sense).